src/server.cpp

```diff
@@ -1119,16 +1119,15 @@ PlayerSAO* Server::StageTwoClientInit(u16 peer_id)
 	// Send inventory
 	SendInventory(playersao);
 
-	// Send HP
-	SendPlayerHPOrDie(playersao);
+	// Send HP or death screen
+	if (playersao->isDead())
```

Comment:
deaths are handled differently for units and players, so moving m_dead into unitSAO and therefore removing getDeadStatus might complicate things? It still should be done though.